Test with ESP8285, ESP32

On ESP8285, simplest sketch (ranging_basic) seems not to fit into memory:

Arduino: 1.8.13 (Linux), TD: 1.53, Board: "Generic ESP8266 Module, 80 MHz, Flash, Legacy (new can return nullptr), All SSL ciphers (most compatible), dtr (aka nodemcu), 26 MHz, 40MHz, DOUT (compatible), 1MB (FS:64KB OTA:~470KB), 2, nonos-sdk 2.2.1+100 (190703), v2 Lower Memory, Disabled, None, Only Sketch, 115200"











/home/levy/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/2.5.0-4-b40a506/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld: ranging_basic.ino.elf section `.rodata' will not fit in region `dram0_0_seg'
/home/levy/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/2.5.0-4-b40a506/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld: address 0x40005378 of ranging_basic.ino.elf section `.bss' is not within region `dram0_0_seg'
/home/levy/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/2.5.0-4-b40a506/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld: address 0x40005378 of ranging_basic.ino.elf section `.bss' is not within region `dram0_0_seg'
collect2: error: ld returned 1 exit status
exit status 1
Error compiling for board Generic ESP8266 Module.


This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.

Are you selecting the "Generic ESP82*85* board" from the board manager?

On Sat, Aug 28, 2021 at 9:31 AM Simon D. Levy ***@***.***>
wrote:

> On ESP8285, simplest sketch (ranging_basic) seems not to fit into memory:
>
> Arduino: 1.8.13 (Linux), TD: 1.53, Board: "Generic ESP8266 Module, 80 MHz,
> Flash, Legacy (new can return nullptr), All SSL ciphers (most compatible),
> dtr (aka nodemcu), 26 MHz, 40MHz, DOUT (compatible), 1MB (FS:64KB
> OTA:~470KB), 2, nonos-sdk 2.2.1+100 (190703), v2 Lower Memory, Disabled,
> None, Only Sketch, 115200"
>
> /home/levy/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/2.5.0-4-b40a506/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld:
> ranging_basic.ino.elf section .rodata' will not fit in region dram0_0_seg'
> /home/levy/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/2.5.0-4-b40a506/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld:
> address 0x40005378 of ranging_basic.ino.elf section .bss' is not within
> region dram0_0_seg'
> /home/levy/.arduino15/packages/esp8266/tools/xtensa-lx106-elf-gcc/2.5.0-4-b40a506/bin/../lib/gcc/xtensa-lx106-elf/4.8.2/../../../../xtensa-lx106-elf/bin/ld:
> address 0x40005378 of ranging_basic.ino.elf section .bss' is not within
> region dram0_0_seg'
> collect2: error: ld returned 1 exit status
> exit status 1
> Error compiling for board Generic ESP8266 Module.
>
> This report would have more information with
> "Show verbose output during compilation"
> option enabled in File -> Preferences.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/simondlevy/VL53L5/issues/5#issuecomment-907650422>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTDLKVR4G7I436JRKWA5KDT7EFNFANCNFSM5CSJEMKQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

Hello Kris,
just got my Range Finder, but the driver is mocking me. 
I am using a ESP32 ("ESP32 dev Module") with Arduino IDE 1.8.9.

------------------------------------

With i2c Address 0x29 and example "get_at_params" i get this in my Serial monitor:
**"WrMulti failed to send 32768 bytes to regsiter 0x00"**

a google search is pointing me to this Forum:
[ST forum - vl53l5cx-driver-vl53l5cxinit-fails](https://community.st.com/s/question/0D53W000010uFcTSAU/vl53l5cx-driver-vl53l5cxinit-fails)

.. there is a suggestion to modify the platform.cpp with code from that forum, so i did as shown there.

Error in "platform.cpp":
>>>>
   **platform.cpp:225:14: error: 'delay' was not declared in this scope
   delay(TimeMs);**
<<<<

so i #include "Arduino.h" to platform.cpp

Error in "platform.cpp":
 all Wire.x got errors >>>>
**platform.cpp:164:3: error: 'Wire' was not declared in this scope
   Wire.requestFrom(((uint8_t)(((p_platform->address) >> 1) & 0x7F)), size);**
<<<<

so i #include "Wire.h" in platform.cpp

Error in "get_set_params.ino":
>>>>
**get_set_params.ino:42: undefined reference to `Reset_Sensor(unsigned char)'
collect2.exe: error: ld returned 1 exit status**
<<<<

So there i replaced in "get_set_params.ino" Reset_Sensor(LPN_PIN) 

with:
    // Reset the sensor by toggling the LPN pin
  digitalWrite(LPN_PIN, LOW);
  delay(100);
  digitalWrite(LPN_PIN, HIGH);
  delay(100);

now Serial monitor tells me: 

"**VL53L5CX not detected at requested address**"  .. even its detectable in i2c scanner

i change the address to 0x52 in "get_set_params.ino"  &  "WL53L5cx.h" to and the serial monitor says no more. Same with 0x53.


I hope any of those informations are somehow useful. If you have any idea where i could investigate further, please let me know.

Best wishes, Julius

All I can tell you Julius is that neither I nor Simon have been able to get
this to work using the ESP32 or ESP8285. There is something not compatible
with the ESPXX  I2C library, don't know what. Someone will figure it out
someday...but today is not that day. You can try a Ladybug or a Teensy in
the meantime; these work.

On Thu, Sep 2, 2021 at 1:51 PM jjoeff ***@***.***> wrote:

> Hello Kris,
> just got my Range Finder, but the driver is mocking me.
> I am using a ESP32 ("ESP32 dev Module") with Arduino IDE 1.8.9.
> ------------------------------
>
> With i2c Address 0x29 and example "get_at_params" i get this in my Serial
> monitor:
> *"WrMulti failed to send 32768 bytes to regsiter 0x00"*
>
> a google search is pointing me to this Forum:
> ST forum - vl53l5cx-driver-vl53l5cxinit-fails
> <https://community.st.com/s/question/0D53W000010uFcTSAU/vl53l5cx-driver-vl53l5cxinit-fails>
>
> .. there is a suggestion to modify the platform.cpp with code from that
> forum, so i did as shown there.
>
> Error in "platform.cpp":
>
>
> *platform.cpp:225:14: error: 'delay' was not declared in this scope
> delay(TimeMs);*
> <<<<
>
> so i #include "Arduino.h" to platform.cpp
>
> Error in "platform.cpp":
> all Wire.x got errors >>>>
>
> *platform.cpp:164:3: error: 'Wire' was not declared in this scope
> Wire.requestFrom(((uint8_t)(((p_platform->address) >> 1) & 0x7F)), size);*
> <<<<
>
> so i #include "Wire.h" in platform.cpp
>
> Error in "get_set_params.ino":
>
>
> *get_set_params.ino:42: undefined reference to `Reset_Sensor(unsigned
> char)' collect2.exe: error: ld returned 1 exit status*
> <<<<
>
> So there i replaced in "get_set_params.ino" Reset_Sensor(LPN_PIN)
>
> with:
> // Reset the sensor by toggling the LPN pin
> digitalWrite(LPN_PIN, LOW);
> delay(100);
> digitalWrite(LPN_PIN, HIGH);
> delay(100);
>
> now Serial monitor tells me:
>
> "*VL53L5CX not detected at requested address*" .. even its detectable in
> i2c scanner
>
> i change the address to 0x52 in "get_set_params.ino" & "WL53L5cx.h" to and
> the serial monitor says no more. Same with 0x53.
>
> I hope any of those informations are somehow useful. If you have any idea
> where i could investigate further, please let me know.
>
> Best wishes, Julius
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/simondlevy/VL53L5/issues/5#issuecomment-912046652>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTDLKVZQEKIMQE7R6UP7UTT77PVJANCNFSM5CSJEMKQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

--- a/src/VL53L5cx.cpp
+++ b/src/VL53L5cx.cpp
@@ -111,7 +111,10 @@ void VL53L5cx::init(void)
     if(error) {
         Debugger::reportForever("VL53L5CX ULD Loading failed");
     }
-
+   
+   // Set the range frequency
+   vl53l5cx_set_ranging_frequency_hz(&_dev, _ranging_frequency);
+   
     // Set resolution
     vl53l5cx_set_resolution(&_dev,
             _resolution == RESOLUTION_4X4 ?
@@ -164,7 +167,7 @@ uint8_t VL53L5cx::getTargetStatus(uint8_t zone, uint8_t target)
     return _results.target_status[VL53L5CX_NB_TARGET_PER_ZONE * zone + target];
 }
 
-uint8_t VL53L5cx::getDistance(uint8_t zone, uint8_t target)
+int16_t VL53L5cx::getDistance(uint8_t zone, uint8_t target)
 {
     return _results.distance_mm[VL53L5CX_NB_TARGET_PER_ZONE * zone + target];
 }

--- a/src/VL53L5cx.h
+++ b/src/VL53L5cx.h
@@ -68,7 +68,7 @@ class VL53L5cx {
 
         uint8_t getTargetStatus(uint8_t zone, uint8_t target=0);
 
-        uint8_t getDistance(uint8_t zone, uint8_t target=0);
+        int16_t getDistance(uint8_t zone, uint8_t target=0);
 
         uint8_t getSignalPerSpad(uint8_t zone, uint8_t target=0);
 

